### ISSUE TITLE
Allow copy of existing objects in form ?

### ISSUE THREAD

**Issue by glennpierce** [eclipsesource/jsonforms-vuetify-renderers#59]
I have a json schema like  https://pastebin.com/XHY4fkLT

It works ok but it has a hierarchy of objects

ie an array of Sites contains an array of Equip which contains an array of points

[Sites] -> [Equip] -> [Points]

The equipment and point entries probably have very similar entries to all the others

What I would like is that twhen the user adds a new site with the + button all the child fileds of the previous added site get filled in automatically ?

Or

along side each site there is a copy button nect to the (delete bin icon) that copies that entry.

Basically I need to make it easier/quicker for the user to add complex objects. Is there a way to do it currently ?

Thanks

**Comment by sdirix** [comment]
Hi! Please use our [discussion board for questions](https://jsonforms.discourse.group/).

I can think of these approaches:

1. **Implement custom renderers**: By using custom renderers you can implement yourself what shall happen when a user clicks the "Plus" button.
2. **Modify from the outside**: You can observe the data changes of the `json-forms` component and whenever such a new (empty) object is added you modify the data to add the attributes you want.
3. **Depending on your use case: Use the default mechanism of AJV**: Internally we use AJV for validation. While validating AJV can fill in `default` values of the JSON Schema into the data. So if that is sufficient for your use cases, see [here for more information](https://jsonforms.discourse.group/t/help-with-default-values/31/2).
